### ISSUE TITLE
Expanding an item that is partially obscured by the top of the scrolling window causes the item to move upwards

### ISSUE THREAD

**Issue by adlai-arnold** [TanStack/virtual#562]
### Describe the bug

Expanding an item should never move the top of the item--it should always expand downwards. However, if the scroll window occludes the top of the item, it will instead move upwards when expanded. 

**This issue does not occur in `3.0.0-beta.35` and earlier.**

### Your minimal, reproducible example

https://codesandbox.io/p/sandbox/zen-blackburn-tnrzfr

### Steps to reproduce

1. Scroll down slightly, so that the first item is partially occluded by the top of the scroll window
2. Click the first item to expand it
3. Note that the top of the item jumps up and out of view

### Expected behavior

As a user, I expect that expanding an item will not cause it to move.

### How often does this bug happen?

Every time

### Platform

MacOS 13.4.1 (22F82), Chrome Version 114.0.5735.198 (Official Build) (arm64)

### tanstack-virtual version

3.0.0-beta.54

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by adlai-arnold** [comment]
Given that the issue starts in beta release 36, it must be somewhere in this diff: https://github.com/TanStack/virtual/compare/v3.0.0-beta.35...v3.0.0-beta.36

**Comment by reidark** [comment]
Any solutions/workaround on that? 

I have a virtual list of checkboxes and every time the user select some item that is not 100% on the "viewport", it scrolls upwards, causing a confusion.

This happens because of the focus state? I know if I navigate via keyboard the list will scroll automatically based on the element focused.

**Comment by piecyk** [comment]
No it is happening because of our auto adjustment scroll position logic, basic if user jumps to bottom of the list an start scrolling up, in dynamic case the sizes can change, that will change the height of the list making the scroll to jump


https://github.com/TanStack/virtual/blob/beta/packages/virtual-core/src/index.ts#L656-L665

with this we are trying to limit that behaviour, best would to be to add it only when scrolling backward and maybe to allow to opt out from it.

**Comment by kamuiiiii** [comment]
I've encountered the same issue. Any updates on this?

**Comment by piecyk** [comment]
> I've encountered the same issue. Any updates on this?

Yeah, now it's possible to control the adjustment with `shouldAdjustScrollPositionOnItemSizeChange`, one option can be 

```jsx
const expand = (index: number) => {
  virtualizer.shouldAdjustScrollPositionOnItemSizeChange = () => false
  setExpanded(index)
  // move to next tick after resize observable will update sizes
  setTimeout(() => {
    virtualizer.shouldAdjustScrollPositionOnItemSizeChange = undefined
  }, 0)
} 
```
